gorfc/gorfc.go

```diff
@@ -1216,3 +1221,41 @@ func (conn *Connection) Call(goFuncName string, params interface{}) (result map[
 	}
 	return wrapResult(funcDesc, funcCont, C.RFC_IMPORT, conn.rstrip)
 }
+
+func (conn *Connection) rfcCancel(goFuncName string) error {
+	var errorInfo C.RFC_ERROR_INFO
+	rc := C.RfcCancel(conn.handle, &errorInfo)
+	if rc != C.RFC_OK {
+		return rfcError(errorInfo, "Could not invoke function \"%v\"", goFuncName)
+	}
+	conn.alive = false
+	return nil
+}
+
+// Call calls the given function with the given parameters and wraps the results returned.
+func (conn *Connection) Call(goFuncName string, params interface{}) (result map[string]interface{}, err error) {
+	return conn.CallContext(context.Background(), goFuncName, params)
+}
+
+// CallContext calls the given function with the given parameters and wraps the results returned.
+func (conn *Connection) CallContext(ctx context.Context, goFuncName string, params interface{}) (result map[string]interface{}, err error) {
+	if ctx.Done() == nil {
+		return conn.rfcInvoke(goFuncName, params)
+	}
+
+	done := make(chan interface{})
+	go func() {
+		defer close(done)
+		result, err = conn.rfcInvoke(goFuncName, params)
+		done <- true
+	}()
+
+	select {
+	case <-ctx.Done():
+		conn.alive = false
+		err = errors.Join(ctx.Err(), conn.rfcCancel(goFuncName), conn.Open())
```

Comment:
Minor suggestion: Could we add global errors here? 

This would let anyone to use `errors.Is(X)` to check for things like a canceled context, instead of checking strings. 

For example, `CallContext` could return an error wrapped with `rfcCancel`, making error assertions easier

Reply:
Good point. I added an assertion to the test.